src/ssl.c

```diff
@@ -54563,65 +54563,96 @@ int wolfSSL_BN_rand(WOLFSSL_BIGNUM* bn, int bits, int top, int bottom)
     WC_RNG        tmpRNG[1];
     byte          buff[1024];
 #endif
+    int           lastBits = 0;
 
-    (void)top;
-    (void)bottom;
-    WOLFSSL_MSG("wolfSSL_BN_rand");
+    WOLFSSL_ENTER("wolfSSL_BN_rand");
 
-    if (bits <= 0) {
-        return WOLFSSL_FAILURE;
+    if ((bn == NULL || bn->internal == NULL) || bits < 0 ||
+        (bits == 0 && (bottom != 0 || top != -1)) || (bits == 1 && top > 0)) {
```

Comment:
I don't see requesting a number with 0 bits as useful.
Does OpenSSL support this?

Reply:
OpenSSL supports this, yes.